### DISCUSSION TITLE
Does Supabase support the new email/password verification link logic like Firebase?

### DISCUSSION THREAD

**Discussion by davoutuk** [supabase/supabase#11590] [Questions] [answered]
I am assessing Supabase as a replacement for Firebase.

On area I am struggling with is the use of 'verification' emails to validate a new user.  With Firebase,  if you create a new email/password based user then Firebase will send out a 'verification' email to that email address.   Inside the email is a link to a Firebase URL that when clicked will action the verification logic.

I can't find the equivalent logic with Supabase.  I can see from the management console that you can define a custom verification URL, but I can't any docs to explain how this verification acceptance logic can be handled by Supabase directly.

This seems like a strange gap.  Am I supposed to create a custom URL myself and then execute logic to perform the verification acceptance process?

**Comment by GaryAustin1** [comment, marked as answer]
This explains one of the signup calls for javascript:  https://supabase.com/docs/reference/javascript/auth-signup

An email is sent and it has a link that goes back to supabase to verify the user.  Then supabase will call your app again.  You can use redirectTo option to specify a path for that call back.  Your user will be signed by the client initialization code and a token in that redirect.

You can also use a 6 digit code in the email to verify the user.

**Reply by GaryAustin1** [reply]
It is better to reply to answers than create new answers, if they are not a solution.

I think what you want to do is built in.  
This is a template that by default sends the supabase verification link.
![image](https://user-images.githubusercontent.com/54564956/211860692-ddfe7836-1247-4949-9c53-709070ed0145.png)
And these settings give the option to do send that template...

![image](https://user-images.githubusercontent.com/54564956/211861102-5dc2c0d1-cfb8-46f9-9ff5-fada5160b50b.png)

Or are you wanting to add your own verification process?

**Reply by davoutuk** [reply]
Here is my scenario...

- I'm building a mobile app that will use Supabase, with the 'confirm email' active for new sign ups.
- The mobile app will have a 'sign up' screen that will under the hood call the  'Supabase.instance.client.auth.signUp' method
- Supabase will send out the email with the verification link
- The user can pick up the email on any device, and click on the verification link.  In my case I am viewing the email on my Windows PC in Outlook
- Clicking on the verification acceptance link would I presume have to call some central Supabase URL, as their is no guarantee that the same device was used for new user sign up and email verification acceptance actions.

In that light, how and where can I call a Supabase function on a Supabase server that handle the verification acceptance call?

**Reply by GaryAustin1** [reply]
Your steps all seem OK .  When you click on that url in the email it will open up a browser on whatever device that user on and go to the supabase url, then the browser will be redirected back to / or another page provided in the redirectTo option of the signup call.  Some people use a "deep link" to get back to their app (assumes email client on same device).  Some have a web page for the initial redirect landing and then call a deep link from that.. https://github.com/supabase/supabase/discussions/7093#discussioncomment-2944265

So your user will get verified with that link and then the issue is the flow back to your app (or tell them to go back to your app) and sign them in (the verification was already sent to the server so they will sign in.)  But without the token in the redirect URL the supabase client can't signin automatically.   

I do a home screen web app and this flow is not acceptable for my use.  So I use the 6 digit OTP method and leave my app in a screen to enter the code that shows up in the email.  When the user enters that code, I call verifyOTP which then finishes the signup and signs in the user, all staying in the app.

Sorry if I am not answering your question exactly.  There are discussions on native apps and oath here and in issues both at supabase/supabase and supabase/gotrue.  I bailed on the email link path and went with the OTP method.  Maybe someone else has something for you.

**Reply by GaryAustin1** [reply]
Adding the REST API call that verifies the url token sent in the email (this is what the url in the email initially calls)...
https://github.com/supabase/gotrue#post-verify

**Reply by davoutuk** [reply]
Thanks for the feedback...

How is the 'siteURL' supposed to work when I have a Flutter app that can be both a web app and an Android app?

**Reply by GaryAustin1** [reply]
You use redirectTo in your calls and set differently for each environment.  You also have to add the full URL in additional URL setting.
